### ISSUE TITLE
Pace split

### ISSUE THREAD

**Issue by FlorianDeconinck** [GEOS-ESM/pace#41]
`Pace` will be split alongside three axis and recomposed from them.
- `ndsl`: middleware part
- `PyFV3`: the Python/DSL version of FV3
- `PySHiELD`: the Python/DSL version of physics parametrization

`Pace` remains a model, bringing the three parts above as submodules and dealing with driver & bespoke init.

⚠ Due to the breath of the refactor - a coding freeze is required⚠ 

Branch/PR wise, a freeze will be enacted then the following pattern will be implemented.
```
main (freeze) |----|----|----|---|--------|--- main (resume dev)
               \   /    \   /    \       /     
	       ndsl     pyfv3    pySHiELD
```

In-between each merge, a full fresh install of Pace + running a C12 for 1 timesteps will insure functionality.

We will document here all changes. Post refactor, we expect some movement of code from and to the new repository but that should be minimal.

Out of scope for this issue.
- `makefile` workflow
- `docker` image
- `github` CI hooks
- `docs`

DoD: Pace relies on `ndsl`, `pyFV3` and `pySHiELD`.

Cross-post: https://github.com/NOAA-GFDL/pace/issues/53

**Comment by FlorianDeconinck** [comment]
`ndsl`
- `tracer_variables`: need an API in `ndsl` and calls in the model. Stays in `ndsl` for now, API needed.
- `halo` and `origin` in `gt4py_utils`: those should have an API with reasonable defaults.

`physics`
- `update_dwind_phys.py`, `update_atmos_state.py` and `fv_update_phys.py` lives in Physics but should be part of a broader plug & play Physics API that know how to interface with the Dynamical Core.

**Comment by FlorianDeconinck** [comment]
`ndsl`
- `examples/mpi` are the examples from util

**Comment by FlorianDeconinck** [comment]
`ndsl`
- There's a C12 data bundle for testing restart in `tests/main/data` on Pace

**Comment by FlorianDeconinck** [comment]
`ndsl`
- Clean up the `__init__.py` of the package or subdirectory to only publicize the correct API. Or do away entirely with the concept.
- `utils`:
  - Halo updates are now bundled in `utils.halo`
  - Comms, partition & MPI are bundled `utils.comm`

Todo:
- Reconnect `utils` tests to github workflow

**Comment by FlorianDeconinck** [comment]
`ndsl`
- `utils`:
  - remove `_corners`, `_capture_streams`, `_xarray`, `nudging`, `time`
  - move `util._timing` in `performance`
  
 Todo:
 - `PACE_LOGLEVEL`: `NDSL_LOGLEVEL`
 - `restart` should move to Pace and/or pyFV3

**Comment by FlorianDeconinck** [comment]
`ndsl`

- Do we keep the `__init__.py` has a method of public API (reduce the import on top of the files) or keep the full import structure (less cross imports)

**Comment by FlorianDeconinck** [comment]
`ndsl`

- `pre-commit` setup

**Comment by FlorianDeconinck** [comment]
## Porting errors to fix

- `ndsl.grid.eta` depends on pyFV3 for ETA calculation.
- `ndsl.stencils.c2l_ord` depends on pyFV3, requires DycoreState

## Breakage due to refactor
- `docs` is broken and need rewire
- `Makefile` are broken in all new repositories
- `docker` are broken in all new repositories

**Comment by FlorianDeconinck** [comment]
`todo`

- close issues
- x-post to NOAA
- set up working branches and secure `main`

**Comment by FlorianDeconinck** [comment]
`todo`

- makes issues
- x-post to NOAA
- set up working branches and secure `main`

**Comment by FlorianDeconinck** [comment]
New repositories:
[ndsl](https://github.com/NOAA-GFDL/NDSL)
[pyFV3](https://github.com/NOAA-GFDL/pyFV3)
[pySHiELD](https://github.com/NOAA-GFDL/pySHiELD)
